internal/terraform/module/module_ops.go

```diff
@@ -293,8 +294,21 @@ func ParseModuleManifest(fs filesystem.Filesystem, modStore *state.ModuleStore,
 		}
 		return err
 	}
+	if mm != nil {
+		for _, modRecord := range mm.Records {
+			modModulePath := path.Join(modPath, modRecord.Dir)
+			_, err := modStore.ModuleByPath(modModulePath)
+			if err != nil {
+				modStore.Add(modModulePath)
+				ParseModuleConfiguration(fs, modStore, modModulePath)
+				ParseVariables(fs, modStore, modModulePath)
+				LoadModuleMetadata(modStore, modModulePath)
+			}
+		}
+	}
```

Comment:
This looks functionally reasonable and I understand that the operations are all dependent on the manifest being parsed, but I worry slightly that this (a) stretches the responsibility of `ParseModuleManifest` and (b) introduces potential bottleneck in the queue processing logic 🤔 

I will have a look if there's any way of introducing some simple dependency support via callbacks to the scheduler - so that we can just get this scheduled upon completion of `ParseModuleManifest` but still have it fall into the same queue as all other operations instead of synchronously executing it here.

Ideally we'd then be able to move such "dependent scheduling" to the watcher and walker and not have to introduce more responsibilities to this operation.

Reply:
Yeah I was not that happy with this too but not so sure hot to schedule this too. by the way this should be using filepath instead of path to join too will add that.

Reply:
I am also trying to execute the add the minimum amount of time possible so it will only execuute on the first pass of a module not on all occurencies.

Reply:
I will address this in a separate follow up PR, which will also take advantage of https://github.com/hashicorp/terraform-ls/pull/590 - your PR looks good otherwise.